tests/tests_pytorch/callbacks/test_quantization.py

```diff
@@ -109,7 +109,7 @@ def test_quantize_torchscript(tmpdir):
     trainer = Trainer(callbacks=[qcb], default_root_dir=tmpdir, max_epochs=1)
     trainer.fit(qmodel, datamodule=dm)
 
-    batch = iter(dm.test_dataloader()).next()
+    batch = next(iter(dm.test_dataloader()))
```

Comment:
Isn't it strange that we had to fix this in this PR? This failed in the PyTorch 1.13 jobs. Surprised that `.next()` even worked, have never seen that.

Reply:
yep, because these tests were not running before.

Reply:
Do you know how they were skipped? The only marker on that test is quantization=True and I don't see a connection to the changes in this PR

Reply:
yep: https://github.com/Lightning-AI/lightning/blob/773cb3e8c8d08ea373638d40be0831e8b28ed937/tests/tests_pytorch/helpers/datamodules.py#L112-L115

Reply:
should we add sklearn to runif?

Reply:
Oh wow, I smell that this is a bad practice to hide a pytest.skip like this. 

> should we add sklearn to runif?

If we have enough usages that justify it, yes I think so. Otherwise, at least we should put the skip mark on the test itself.

Reply:
yep agreed! component should raise an error, pytest should handles skips